spec/features/create_etd_spec.rb

```diff
@@ -0,0 +1,235 @@
+    expect(page.find('#pbAbstractProvided')['style']).to match(/background-image/)
+
+    # the hydra_etd app has all the <input type=file> tags at the bottom of the page, disabled,
+    #   and when uploading files, we have to attach the file to the right one of these elements
+    #   I think this may be an artifact of the js framework it usees, prototype
+    file_upload_elements = all('input[type=file]', visible: false)
+
+    # upload dissertation PDF
+    filename = 'etd_dissertation.pdf'
+    expect(page.find('#pbDissertationUploaded')['style']).to eq '' # dissertation PDF not yet provided
+    expect(page).not_to have_content(filename)
+    dissertation_pdf_upload_input = file_upload_elements.first
+    dissertation_pdf_upload_input.attach_file("spec/fixtures/#{filename}")
+    sleep(3) # wait for upload
+    expect(page).to have_content(filename)
+    # a checked box in the progress section is a background image
+    expect(page.find('#pbDissertationUploaded')['style']).to match(/background-image/)
+
+    # upload supplemental file
+    find('input#cbSupplementalFiles').check
+    filename = 'etd_supplemental.txt'
+    expect(page).not_to have_content(filename)
+    # supplemental files uploaded progress checkbox not visible by default
+    expect(page.find('#pbSupplementalFilesUploaded', visible: false)).not_to be_visible
+    supplemental_upload_input = file_upload_elements[1]
+    supplemental_upload_input.attach_file("spec/fixtures/#{filename}")
+    expect(page).to have_content(filename)
+    # a checked box in the progress section is a background image
+    expect(page.find('#pbSupplementalFilesUploaded')['style']).to match(/background-image/)
+
+    # indicate copyrighted material
+    expect(page.find('#pbPermissionsProvided')['style']).to eq '' # rights not yet selected
+    page.select 'does include', from: 'selectPermissionsOptions'
+
+    # provide copyright permissions letters/files
+    # permission files uploaded progress checkbox not visible by default
+    expect(page.find('#pbPermissionFilesUploaded', visible: false)).not_to be_visible
+    filename = 'etd_permissions.pdf'
+    expect(page).not_to have_content(filename)
+    permissions_upload_input = file_upload_elements[11]
+    permissions_upload_input.attach_file("spec/fixtures/#{filename}")
+    expect(page).to have_content(filename)
+    # a checked box in the progress section is a background image
+    expect(page.find('#pbPermissionsProvided')['style']).to match(/background-image/)
+    expect(page.find('#pbPermissionFilesUploaded')).to be_instance_of Capybara::Node::Element
+
+    # apply licenses
+    expect(page.find('#pbRightsSelected')['style']).to eq '' # rights not applied yet
+    click_link('View Stanford University publication license')
+    page.find('input#cbLicenseStanford').check
+    click_link('Close this window')
+    click_link('View Creative Commons licenses')
+    page.select 'CC Attribution license', from: 'selectCCLicenseOptions'
```

Comment:
Page should be unnecessary

Reply:
true.  I will remove.  Doesn't seem like a big deal, tho.